astroid/nodes/scoped_nodes/scoped_nodes.py

```diff
@@ -1495,7 +1497,10 @@ def is_generator(self) -> bool:
 
         :returns: Whether this is a generator function.
         """
-        return bool(next(self._get_yield_nodes_skip_lambdas(), False))
+        yields_without_lambdas = set(self._get_yield_nodes_skip_lambdas())
+        yields_without_functions = set(self._get_yield_nodes_skip_functions())
+        # Want an intersecting member that is neither in a lambda nor a function
+        return bool(yields_without_lambdas & yields_without_functions)
```

Comment:
This is what necessitated all those new methods, given that FunctionDefs are no longer lambdas... (lol) we can't just skip lambdas only.